cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/AbstractKubeTwillRunnerService.java

```diff
@@ -77,45 +74,41 @@
  *
  * cdap.twill.app=[literal app name]
  */
-public class KubeTwillRunnerService implements TwillRunnerService {
+abstract class AbstractKubeTwillRunnerService<T> implements TwillRunnerService {
```

Comment:
Why this is needed? We shouldn't have one runner service per resource type.

Reply:
Had to create different versions based on the resource types primarily for object meta which is required in the ResourceWatchListener implementation such as here - https://github.com/cdapio/cdap/blob/develop/cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java#L325

Instead of having different versions of TwillRunnerService, now have `ObjectMetaProvider<T>` interface to provide the object meta.